be/tool/main.c

```diff
@@ -59,6 +76,95 @@ static const char *betool_help = ""
 "Default BE domain stob domain location is used when this parameter is absent:"
 "\n";
 
+extern void btree_dbg_print(struct m0_be_btree *tree);
+
+void track_cob_btrees(struct m0_cob_domain *cdom, bool print_btree)
+{
+	if (print_btree) {
+		M0_LOG(M0_ALWAYS, "cd_object_index ");
+		btree_dbg_print(&cdom->cd_object_index);
+		M0_LOG(M0_ALWAYS, "cd_namespace ");
+		btree_dbg_print(&cdom->cd_namespace);
+		M0_LOG(M0_ALWAYS, "cd_fileattr_basic ");
+		btree_dbg_print(&cdom->
+				cd_fileattr_basic);
+	} else
+		M0_LOG(M0_ALWAYS,"M0_BE:COB "
+				"cd_object_index btree = %p "
+				"cd_namespace btree = %p "
+				"cd_fileattr_basic btree = %p "
+				"cd_fileattr_omg btree = %p "
+				"cd_fileattr_ea btree = %p",
+				&cdom->cd_object_index,
+				&cdom->cd_namespace,
+				&cdom->cd_fileattr_basic,
+				&cdom->cd_fileattr_omg,
+				&cdom->cd_fileattr_ea);
+}
+
+void track_ad_btrees(struct stob_ad_0type_rec *rec, bool print_btree)
+{
+	struct m0_balloc         *m0balloc;
+
+	m0balloc = container_of(rec->sa0_ad_domain->sad_ballroom,
+				struct m0_balloc, cb_ballroom);
+
+	if (print_btree) {
+		M0_LOG(M0_ALWAYS, "em_mapping");
+		btree_dbg_print(&rec->sa0_ad_domain->sad_adata.em_mapping);
```

Comment:
btree_dbg_print() uses the M0_DEBUG trace log level.
It will not print to stdout in default mode.

The user of this util should set trace log level more verbose, or
need to look into the trace file.

Reply:
yes, proper trace log level needs to set,
Can we pass log level as M0_ALWAYS to print btree nodes from m0betool, and  M0_DEBUG  from existing btree code ?